schema/roomdetails/units.ts

```diff
@@ -22,5 +26,370 @@ is each lowest-level administrative division within central services.`,
+											}
+										} else {
+											errors.push(`Person ${person.person.sciper} not found.`)
+										}
+									}
+									else if (person.status == 'Deleted') {
+										let p = await tx.Person.findUnique({ where: { sciper: person.person.sciper }});
+										if (p) {
+											try {
+												const del = await tx.unit_has_cosec.deleteMany({
+													where: {
+														id_unit: unit.id,
+														id_person: p.id_person
+													}
+												});
+												if (!del) {
+													errors.push(`Relation unit-cosec not deleted between ${unit.name} and ${person.person.sciper}.`)
+												}
+											} catch ( e ) {
+												errors.push(`DB error: relation unit-cosec not deleted between ${unit.name} and ${person.person.sciper}.`)
+											}
+										} else {
+											errors.push(`Person ${person.person.sciper} not found.`)
+										}
+									}
+								}
+							}
+
+							if (args.subUnits.length>0) {
+								for (const subunit of args.subUnits) {
+									if (subunit.status == 'New') {
+										try {
+											const u = await tx.Unit.create({
+												data: {
+													name: subunit.name,
+													id_institute: unit.id_institute
+												}
+											});
+											if ( !u ) {
+												errors.push(`Error creating sub-unit ${subunit.name}.`);
+											}
+										} catch ( e ) {
+											errors.push(`Error creating sub-unit ${subunit.name}.`);
+										}
+									}
+									else if (subunit.status == 'Deleted') {
+										const u = await tx.Unit.findFirst({ where: { name: subunit.name }});
+										if (u) {
+											errors.concat(await deleteUnit(tx, u));
+										} else {
+											errors.push(`Error deleting sub-unit ${subunit.name}.`)
+										}
+									}
+								}
+							}
+						} catch ( e ) {
+							errors.push(`Error updating unit.`)
+						}
+
+						if (errors.length > 0) {
+							return mutationStatusType.error(`${errors.join('\n')}`);
+						} else {
+							return mutationStatusType.success();
+						}
+					} else {
+						return mutationStatusType.error(`Unit ${args.unit} not found.`)
+					}
+				});
+			}
+		});
+		t.nonNull.field('deleteUnit', {
+			description: `Delete unit details by unit name (profs, cosecs, sub-units).`,
+			args: {
+				...unitDeleteType
+			},
+			type: "UnitStatus",
+			async resolve(root, args, context) {
+				const prisma = context.prisma;
+				return await prisma.$transaction(async (tx) => {
+					const unit = await tx.Unit.findFirst({ where: { name: args.unit }});
+					if (unit) {
+						const errors: string[] = [];
+						try {
+							errors.concat(await deleteUnit(tx, unit));
+						} catch ( e ) {
+							errors.push(`Error updating unit.`)
+						}
+
+						if (errors.length > 0) {
+							return mutationStatusType.error(`${errors.join('\n')}`);
+						} else {
+							return mutationStatusType.success();
```

Comment:
Only this code path should commit the transaction. Please throw errors otherwise, or find a smarter way.

Reply:
Done: if errors, we need to throw a new exception in the transaction to make prisma do rollback (https://github.com/prisma/prisma/issues/11107). 
The entire transaction is now inside a try catch block to pass the errors to the client with the `mutationStatusType.error(e.message);`